### PR TITLE
feat(aggregations): Update some aggregation builder components to leafygreen / compass components COMPASS-6234

### PR DESCRIPTION
**Pull request by lerouxb**
Progress so far:
Screenshot 2022-12-01 at 15 57 04

### CODE REVIEW

File: packages/compass-components/src/components/spin-loader.tsx [hunk 1]
```diff
@@ -37,4 +49,6 @@ function SpinLoader({ size = '12px', title }: SpinLoaderProps): JSX.Element {
   );
 }
 
+const SpinLoader = withTheme(UnstyledSpinLoader) as typeof UnstyledSpinLoader;
```
Comment (Anemy) [inline packages/compass-components/src/components/spin-loader.tsx:52]: Until that `withTheme` hook is cleaned up a bit, I'm thinking it's probably cleaner to use the `useDarkMode` hook inside of the function.

Comment (lerouxb) [reply packages/compass-components/src/components/spin-loader.tsx:52]: Aah I already forgot we had that

Comment (gribnoysup) [reply packages/compass-components/src/components/spin-loader.tsx]: I did some changes to the spinner in https://github.com/mongodb-js/compass/pull/3708/files#diff-74dca60361c92e5934a3dffeeca1949b3a6e223cdb72e261c4017022af5738f0 for the shell update, but this is blocked now, just FYI if you want to pick them up too

File: packages/compass-aggregations/src/components/stage-editor-toolbar/stage-collapser.spec.jsx [hunk 2]
```diff
@@ -25,15 +24,16 @@ describe('StageCollaper [Component]', function() {
...
+      expect(component.find('StageCollapser')).to.be.present();
     });
 
     it('renders the collapse text', function() {
       expect(component.find('button')).to.have.prop('title', 'Collapse');
     });
 
     it('renders the collapse button', function() {
-      expect(component.find('.fa-angle-down')).to.be.present();
+      console.log(component.debug());
```
Comment (Anemy) [inline packages/compass-aggregations/src/components/stage-editor-toolbar/stage-collapser.spec.jsx:35]:
```suggestion
```

File: packages/compass-aggregations/src/components/loading-overlay/loading-overlay.tsx [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+const loadingOverlayStylesDark = css({
+  backgroundColor: 'black',
+});
+
+const loadingOverlayStylesLight = css({
+  backgroundColor: 'white',
+});
+
+const textStyles = css({
+  marginLeft: spacing[1]
```
Comment (Anemy) [inline packages/compass-aggregations/src/components/loading-overlay/loading-overlay.tsx:33]:
```suggestion
  marginLeft: spacing[2]
```
Looked slightly better to me, how do you feel about it?

File: packages/compass-aggregations/src/components/loading-overlay/loading-overlay.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        darkMode ? loadingOverlayStylesDark : loadingOverlayStylesLight
+      )}
+    >
+      <SpinLoader />
+      <Body className={textStyles}>{text}</Body>
+    </div>
+  );
+}
+
+const LoadingOverlay = withTheme(
```
Comment (Anemy) [inline packages/compass-aggregations/src/components/loading-overlay/loading-overlay.tsx:55]: Let's use the `useDarkMode` hook instead here?

Comment (lerouxb) [reply packages/compass-aggregations/src/components/loading-overlay/loading-overlay.tsx:55]: hmm I thought I changed that.

File: packages/compass-aggregations/src/components/input-documents-count/input-documents-count.jsx [hunk 5]
```diff
@@ -20,14 +22,9 @@ class InputDocumentsCount extends PureComponent {
    * @returns {Component} The component.
    */
   render() {
-    const iconClassName = classnames({
-      'fa': true,
-      'fa-database': true,
-      [ styles['input-documents-count-db'] ]: true
-    });
     return (
       <div className={classnames(styles['input-documents-count'])}>
```
[the remark this answers is not in the document]
Comment (lerouxb) [reply packages/compass-aggregations/src/components/input-documents-count/input-documents-count.jsx:26]: I'll add some for now, but I'm gonna redo this whole input box in the next PR anyway.

**Pull request merged**